resolver/src/lookup.rs

```diff
@@ -35,19 +38,41 @@ use resolver_future::BasicResolverHandle;
 #[derive(Clone, Debug, Eq, PartialEq)]
 pub struct Lookup {
     rdatas: Arc<Vec<RData>>,
+    valid_until: Instant,
 }
 
 impl Lookup {
-    /// Return new instance with given rdatas
-    pub fn new(rdatas: Arc<Vec<RData>>) -> Self {
-        Lookup { rdatas }
+    /// Return new instance with given rdatas and the maximum TTL.
+    pub fn new(rdatas: Arc<Vec<RData>>,) -> Self {
```

Comment:
> I think we should switch every caller to use `new_with_deadline` and have them pass in `valid_until`. That way we don't accidentally default when we shouldn't. Is that practical or is there something that makes that awkward?

I don't think there's anything that makes that particularly difficult. I originally chose to leave the `new` constructor as-is and make it default because I wanted to keep the diff small, but I'm happy to change this so the caller is responsible for defaulting if you think that's better.